### ISSUE TITLE
console.assert is not a function

### ISSUE THREAD

**Issue by gabrielboliveira** [AtomLinter/linter-eslint#941]
## Issue Type
Bug

## Issue Description
When using [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue), I get an error that `console.assert` is not a function, but running it on Atom Dev Tools, this method is available. Also, running the eslint command on cli doesn't trigger this issue.
I've created issue vuejs/eslint-plugin-vue#35, but I believe it's also related to this package.

[vscode-eslint does](https://github.com/Microsoft/vscode-eslint) not trigger the same issue, works as designed.

![console-assert-issue](https://user-images.githubusercontent.com/11093090/27266683-60a1e6ec-5479-11e7-90e5-38080161a5ea.png)

![console-assert-available](https://user-images.githubusercontent.com/11093090/27266693-72c0e580-5479-11e7-9f19-eea3f663f0cc.png)

### Linter Eslint: Debug output
```
Atom version: 1.18.0
linter-eslint version: 8.2.1
ESLint version: 4.0.0
Hours since last Atom restart: 0
Platform: darwin
Using local project ESLint from: /Users/gabrieloliveira/Projetos/web/node_modules/eslint
Current file's scopes: [
  "text.html.vue",
  "source.js.embedded.html",
  "meta.delimiter.object.comma.js"
]
linter-eslint configuration: {
  "lintHtmlFiles": true,
  "rulesToSilenceWhileTyping": [
    "no-trailing-spaces",
    "eol-last"
  ],
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToDisableWhileFixing": []
}
```

**Comment by Arcanemagus** [comment]
Atom (and Electron / VS Code) itself has a fuller set of functionality available to it than the worker process that ESLint is running under, which runs as a process that emulates Node.js... all that said, `console.assert` _should_ still exist in that process, although I'm not sure whether it is piped back to the calling process.

Maybe we can force it into existence in the worker and pipe anything sent to it out on `stderr`? 😕

**Comment by gabrielboliveira** [comment]
@Arcanemagus I agree with you that it should exists in that process, but for some reason the only available method I was able to test was `console.log`, all others are not available. Perhaps its not a linter-eslint issue, but linter itself or even Atom?

**Comment by Arcanemagus** [comment]
It would be an Electron issue if anything. With the upcoming Atom v1.19 release though we will have access to WebWorkers and can abandon that whole thing in favor of proper support.

**Comment by gabrielboliveira** [comment]
@Arcanemagus I managed to fix this issue on eslint-plugin-vue using node assert package, but still console.assert should be available.

**Comment by Arcanemagus** [comment]
I'm going to mark this as closed here then, this is the only report we have ever had of it, and fixing it wouldn't be done here anyway. Thanks for taking the time to report this and fix it in `eslint-plugin-vue`!
